tgui/packages/tgui/interfaces/common/WirePanel/index.tsx

```diff
@@ -0,0 +1,191 @@
+
+import { useBackend } from "../../../backend";
+import { Box, Button, Collapsible, Dimmer, Icon } from "../../../components";
+import type { WireData, WirePanelData } from './type';
+import { WirePanelActions, WirePanelControlLabels, WirePanelControls, WirePanelCoverStatus } from './const';
+import { BooleanLike } from "common/react";
+import { Fragment } from "inferno";
+
+export const WirePanel = (props, context) => {
+  const { act, data } = useBackend<WirePanelData>(context);
+  const { wire_panel } = data;
+
+  const act_wire = (wire_index: number, action) => {
+    act("actwire", { wire_index, action });
+  };
+
+  return (
+    <Box className="wire_panel">
+      {
+        wire_panel.wires.map((wire, i) => {
+          return (
+            <Wire
+              key={i}
+              dm_index={i+1}
+              color_name={wire.color_name}
+              color_value={wire.color_value}
+              act_wire={act_wire}
+              is_cut={wire.is_cut}
+            />
+          ); }
+        )
+      }
+    </Box>
+  );
+};
+
+interface WireProperties {
+  wires: WireData[],
+  act_wire: any
+}
+
+interface WireProps {
+  dm_index: number
+  color_name: string,
+  color_value: string,
+  is_cut: BooleanLike,
+  act_wire: any
+}
+
+const Wire = (data: WireProps) => {
+  const { dm_index, color_name, color_value, is_cut, act_wire } = data;
+  return (
+    <Fragment>
+      <Box className="wire_label" label={color_name} color={color_value}>{color_name}</Box>
+      <Box className="wire_buttons">
+        { !!is_cut && (
+          <RemoteAccessButton
+            icon="route"
+            content={"Mend"}
+            onClick={() => act_wire(dm_index, WirePanelActions.WIRE_ACT_MEND)}
+          />
+        )}
+        { !is_cut && (
+          <>
+            <RemoteAccessButton
+              icon="cut"
+              content={"Cut"}
+              onClick={() => act_wire(dm_index, WirePanelActions.WIRE_ACT_CUT)}
+            />
+            <RemoteAccessButton
+              icon="bolt"
+              content="Pulse"
+              onClick={() => act_wire(dm_index, WirePanelActions.WIRE_ACT_PULSE)}
+            />
+          </>
+        )}
+      </Box>
+    </Fragment>
+  );
+};
+
+interface IndicatorProperties {
+  control_lights: number,
+  hacked_controls: number
+}
+
+const Indicators = (data: IndicatorProperties) => {
```

Comment:
not complete / using yet, focusing on wires. Compressed the data to a single bitfield so implementors can do whatever they prefer with the data in the UI